src/debugger.c

```diff
@@ -165,41 +165,43 @@ static bool hitBreakpoint(int pc, struct breakpoint bp) {
 int  DEBUGGetCurrentStatus(void) {
 
 	SDL_Event event;
-	if (currentPC < 0) currentPC = pc;							// Initialise current PC displayed.
+	if (currentPC < 0) currentPC = pc;                                      // Initialise current PC displayed.
 
-	if (currentMode == DMODE_STEP) {							// Single step before
-		currentPC = pc;											// Update current PC
-		currentPCBank = getCurrentBank(pc);						// Update the bank if we are in upper memory.
-		currentMode = DMODE_STOP;								// So now stop, as we've done it.
+	if (currentMode == DMODE_STEP) {                                        // Single step before
+		if (currentPC != pc || currentPCBank != getCurrentBank(pc)) {   // Ensure that the PC moved
+			currentPC = pc;                                         // Update current PC
+			currentPCBank = getCurrentBank(pc);                     // Update the bank if we are in upper memory.
+			currentMode = DMODE_STOP;                               // So now stop, as we've done it.
+		}
```

Comment:
This is the meat of the change, and other than the other comment, the only code change.  I believe the rest of the changes are of spacing.